plottr/data/qcodes_dataset.py

```diff
@@ -160,7 +161,7 @@ def get_ds_info(ds: 'DataSetProtocol', get_structure: bool = True) -> DataSetInf
     return data
 
 
-def load_dataset_from(path: str, run_id: int) -> 'DataSetProtocol':
+def load_dataset_from(path: str, run_id: int, read_only: bool = False) -> 'DataSetProtocol':
```

Comment:
perhaps switch this around - make the default argument to signify read-only so that when the code needs also write capabilities then it would need to pass read_only=False explicitly?
```suggestion
def load_dataset_from(path: str, run_id: int, read_only: bool = True) -> 'DataSetProtocol':
```